final_project/main.ipynb

```diff
@@ -0,0 +1,8397 @@
+      "\u001b[90m-) Alexkrakovsky edited page Category:Self-published work in commonswiki\u001b[0m\n",
+      "\u001b[90m-) Alexkrakovsky edited page Category:Self-published work missing SDC copyright license in commonswiki\u001b[0m\n",
+      "\u001b[90m-) Alexkrakovsky edited page Category:Media needing categories as of 23 October 2024 in commonswiki\u001b[0m\n",
+      "\u001b[90m-) Alexkrakovsky edited page Category:All media needing categories as of 2024 in commonswiki\u001b[0m\n",
+      "\u001b[90m-) TimWu007 edited page File:Cimerâu (math o gerfluniau) ar waliau Eglwys y Santes Fair yr Wyddgrug - Chimeras in St Mary's Church, Mold, Wales 08.jpg in commonswiki\u001b[0m\n",
+      "\u001b[90m-) Twofivesixbot edited page Q86405382 in wikidatawiki\u001b[0m\n",
+      "\u001b[90m-) TimWu007 edited page File:Cimerâu (math o gerfluniau) ar waliau Eglwys y Santes Fair yr Wyddgrug - Chimeras in St Mary's Church, Mold, Wales 11.jpg in commonswiki\u001b[0m\n",
+      "630) TimWu007 edited page Category:Taken with DJI FC3582 in commonswiki\n",
+      "\u001b[90m-) Mr.Nostalgic edited page File:Woonkeuken achterzijde - Ootmarsum - 20176794 - RCE.jpg in commonswiki\u001b[0m\n",
+      "631) KrBot edited page Q50963954 in wikidatawiki\u001b[93m (bot)\u001b[0m\n",
+      "632) Monkbot edited page Category:Pages using Lang-xx templates in enwiki\u001b[93m (bot)\u001b[0m\n",
+      "\u001b[90m-) Slzbg edited page Discussion:Coupe du monde féminine de football des moins de 20 ans 2022 in frwiki\u001b[0m\n",
+      "633) TimWu007 edited page Category:Taken with DJI Mini Pro 3 in commonswiki\n",
+      "\u001b[90m-) Egilus edited page Parks in lvwiki\u001b[0m\n",
+      "\u001b[90m-) Rémi sim edited page Q130616983 in wikidatawiki\u001b[0m\n",
+      "\u001b[90m-) DE8AH edited page Catégorie:Wiktionnaire:Prononciations manquantes en italien in frwiktionary\u001b[0m\n",
+      "\u001b[90m-) Kvdrgeus edited page Categorie:Woordenlijst Nederlandse Taal in nlwiktionary\u001b[0m\n",
+      "634) SchlurcherBot edited page File:1908 American La France Funkenblitz pic2.JPG in commonswiki\u001b[93m (bot)\u001b[0m\n",
+      "\u001b[90m-) Nouzen edited page Vampire in the Garden in plwiki\u001b[0m\n",
+      "\u001b[90m-) KababyZMinsem edited page Q112514460 in wikidatawiki\u001b[0m\n",
+      "\u001b[90m-) Mitte27 edited page Брусничное (исторический район) in ruwiki\u001b[0m\n",
+      "\u001b[90m-) MenoBot edited page ريتشارد إل. كونولي in arwiki\u001b[0m\n",
+      "\u001b[90m-) Rémih edited page File:View of La Thuile from the La Nouva slope, 2024.jpg in commonswiki\u001b[0m\n",
+      "635) Monkbot edited page Tuber aestivum in enwiki\u001b[93m (bot)\u001b[0m\n",
+      "\u001b[90m-) Al Silonov edited page питон in ruwiktionary\u001b[0m\n",
+      "\u001b[90m-) MenoBot edited page تصنيف:صفحات تستخدم خاصية P373 in arwiki\u001b[0m\n",
+      "636) TimWu007 edited page Category:Taken with DJI FC3582 in commonswiki\n",
+  
```

Comment:
If you use colors to show events statistics etc, we need to describe and list them